website/server/controllers/api-v3/groups.js

```diff
@@ -607,14 +607,9 @@ api.joinGroup = {
       group.leader = user._id; // If new user is only member -> set as leader
     }
 
-    if (group.type === 'party') {
-      // For parties we count the number of members from the database to get the correct value.
-      // See #12275 on why this is necessary and only done for parties.
-      const currentMembers = await group.getMemberCount();
-      group.memberCount = currentMembers + 1;
-    } else {
-      group.memberCount += 1;
-    }
+    // Set group's new memberCount
+    const currentMembers = await group.getMemberCount();
```

Comment:
Hi @lsophie , thanks for the PR. I'm a bit confused about the PR because `group.getMemberCount()` is not currently using transactions, it's manually counting documents from the database. The idea to fix this issue is to remove `group.getMemberCount()` and instead wrap the code in these routes in a mongodb transactions to make sure the group is always updated correctly. See https://mongoosejs.com/docs/transactions.html for a guide on how to use transactions